test/starlark_tests/rules/generate_xcframework.bzl

```diff
@@ -134,6 +137,34 @@ def _create_xcframework(
         ])
         args.extend(["-framework", framework_path])
 
+        if dsyms and framework_path in dsyms:
+            dsym_files = dsyms[framework_path]
+
+            inputs.extend(dsym_files)
+
+            dsym_bundles_set = sets.make()
+
+            for file in dsym_files:
+                bundle_extension = ".framework.dSYM"
+                prefix, ext, _ = file.short_path.partition(bundle_extension)
+                bundle_path = prefix + ext
+
+                sets.insert(dsym_bundles_set, bundle_path)
+
+                outputs.append(actions.declare_file(
+                    paths.join(
+                        paths.dirname(bundle_path),
+                        "dSYMs",
+                        paths.basename(bundle_path),
+                        paths.relativize(file.short_path, bundle_path),
+                    ),
+                ))
+
+            dsym_bundles = sets.to_list(dsym_bundles_set)
+
+            for dsym_bundle in dsym_bundles:
+                args.extend(["-debug-symbols", dsym_bundle])
```

Comment:
🚨 As-is, this will fail to run with `xcodebuild -create-xcframework` due to some undocumented behavior i was observing. As far as I can tell, unlike other inputs to this command, the paths provided to `-debug-symbols` must be absolute physical paths to the bundles on disk – not relative paths nor symlinks. This is problematic in Bazel, again as far as I know, so I may need to move this all out into a script to run in Execution instead of doing the processing in Analysis.

Reply:
ah never mind, I think I got it! 🎉